src/core/change_detection/constants.ts

```diff
@@ -0,0 +1,91 @@
+import { isBlank } from '../../facade/lang';
+
+/**
+ * Describes the current state of the change detector.
+ */
+export enum ChangeDetectorState {
+  /**
+   * `NeverChecked` means that the change detector has not been checked yet, and
+   * initialization methods should be called during detection.
+   */
+  NeverChecked,
+
+  /**
+   * `CheckedBefore` means that the change detector has successfully completed at least
+   * one detection previously.
+   */
+  CheckedBefore,
+
+  /**
+   * `Errored` means that the change detector encountered an error checking a binding
+   * or calling a directive lifecycle method and is now in an inconsistent state. Change
+   * detectors in this state will no longer detect changes.
+   */
+  Errored
+}
+
+
+/**
+ * Describes within the change detector which strategy will be used the next time change
+ * detection is triggered.
+ */
+export enum ChangeDetectionStrategy {
```

Comment:
well this is premature optimisation and doesn't have anything to do with this PR and new feature.

- anyway in this particular enum it would save 4 lines of code, which is not very huge "payload" saving ( not considering the size after mangling and gzip )

```
(function (ChangeDetectionStrategy) {  
    ChangeDetectionStrategy[ChangeDetectionStrategy["OnPush"] = 0] = "OnPush";
    ChangeDetectionStrategy[ChangeDetectionStrategy["Default"] = 1] = "Default";
})(exports.ChangeDetectionStrategy || (exports.ChangeDetectionStrategy = {}));
```

but fell free to send PR, it can be done